.github/workflows/livetest.yml

```diff
@@ -0,0 +1,89 @@
+      # store output if user is allowed in allowed_user job output so it has to be checked in downstream job
+      - name: Check if user does have correct permissions
+        if: contains('admin write', fromJson(steps.user_permission.outputs.data).permission)
+        id: check_user_perm
+        run: |
+          echo "User '${{ github.event.sender.login }}' has permission '${{ fromJson(steps.user_permission.outputs.data).permission }}' allowed values: 'admin', 'write'"
+          echo "::set-output name=allowed_user::true"
+
+    outputs:
+      allowed_user: ${{ steps.check_user_perm.outputs.allowed_user }}
+
+  live-test:
+    needs: pr-info
+    if: needs.pr-info.outputs.allowed_user == 'true' && github.event.issue.pull_request && startsWith(github.event.comment.body, '/livetest')
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y cockpit-system cockpit-ws nodejs npm sassc make
+
+      - name: Clone repository
+        uses: actions/checkout@v2
+        with:
+          # need this to also fetch tags
+          fetch-depth: 0
+
+      - name: Compile and install the project
+        run: make && sudo make install
+
+      - name: Create a new test user
+        # WARNING: if you want to add this users to sudoers make sure to propertly delete the SINK_SSHRELAY_PRIVATE
+        run: sudo useradd livetest; echo livetest:foobar | sudo chpasswd
+
+      - name: Get ssh gateway port to forward to
+        id: remote_port
+        env:
+          ISSUE_NUMBER: "${{ github.event.issue.number }}"
+        run: |
+            echo ::set-output name=SSH_GATEWAY_PORT::$(($ISSUE_NUMBER % 100 + 15000))
+
+      - name: Write a comment on the PR with link to the cockpit instance
+        uses: actions/github-script@v3
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            github.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: 'Cockpit preview: https://logs.cockpit-project.org:${{ steps.remote_port.outputs.SSH_GATEWAY_PORT }}'
+            })
+
+      - name: Create an SSH private key from the secret
+        run: |
+          sudo touch /run/SINK_SSHRELAY_PRIVATE && sudo chown $(id -u):$(id -g) /run/SINK_SSHRELAY_PRIVATE
+          echo "${{ secrets.SINK_SSHRELAY_PRIVATE }}" > /run/SINK_SSHRELAY_PRIVATE
+          chmod 600 /run/SINK_SSHRELAY_PRIVATE
+          mkdir -p ~/.ssh &&  ssh-keyscan -H logs.cockpit-project.org > ~/.ssh/known_hosts
+
+      - name: Forward cockpit port 9090 to a server we can access
+        run: ssh -f -R ':${{ steps.remote_port.outputs.SSH_GATEWAY_PORT }}':localhost:9090 -i /run/SINK_SSHRELAY_PRIVATE -o ExitOnForwardFailure=yes sshgw@logs.cockpit-project.org 'sleep infinity'
+
+      - name: Block workflow from finishing
+        run: sleep 4h
```

Comment:
FTR: This actually works well even with cancelling -- I just did that, and the workflow *did* run the "cleanup ssh" step, and I just checked on the sink that there are no leftover processes. :+1:

Reply:
Super! Problably as a followup we would like to add a way for the user to cancel that, in case the PR was repushed and they want to regenerate the preview.

Reply:
That shouldn't collide, though? You can easily have many workflows running in parallel. That's a waste, but not broken.

Reply:
Sure, but only one ssh tunnel will be active in the port that the PR corresponds to with the current implementation.

Reply:
Oh, of course, the port would collide. This can be fixed later by not taking the issue number, but the commit SHA % 32768 (so that it fits into the available port range and uses a wide range).

Reply:
I will make this adjustment now - it makes sense that we don't expect the user to stop the workflow to start a new one.